Services/Simulations.cs

```diff
@@ -100,38 +100,53 @@ public async Task<IList<Models.Simulation>> GetListAsync()
         public async Task<Models.Simulation> GetAsync(string id)
         {
             var item = await this.storage.GetAsync(STORAGE_COLLECTION, id);
-            var simulation = JsonConvert.DeserializeObject<Models.Simulation>(item.Data);
-            simulation.ETag = item.ETag;
-            return simulation;
+
+            if (item != null)
+            {
+                var simulation = JsonConvert.DeserializeObject<Models.Simulation>(item.Data);
+                simulation.ETag = item.ETag;
+                return simulation;
+            }
+
+            return null;
         }
 
         /// <summary>
         /// Create a simulation.
         /// </summary>
         public async Task<Models.Simulation> InsertAsync(Models.Simulation simulation, string template = "")
         {
+            var usingDefaultTemplate = !string.IsNullOrEmpty(template) && template.ToLowerInvariant() == "default";
+
             // TODO: complete validation
             if (!string.IsNullOrEmpty(template) && template.ToLowerInvariant() != "default")
```

Comment:
change if condition to if (usingDefaultTemplate)

Reply:
This condition is not the same as usingDefaultTemplate.  There is a suttle difference, we want to exclude the case where template is not provided.